### PR TITLE
add lookup functions for item ID to artifactmap

### PR DESCRIPTION
**Pull request by DnlLrssn**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](https://github.com/qlik-oss/gopherciser/blob/master/.github/CONTRIBUTING.md#commit)
- [ ] documentation updated

**Squash commit message**

Add artifactmap lookup functions also for item ID.

**Info**

Optional informative text (not to be included in commit message) relevant to reviewers.

### CODE REVIEW

File: session/artifactmap.go [hunk 1]
```diff
@@ -135,24 +143,29 @@ func (am *ArtifactMap) FillArtifacts(item *ItemData) error {
 
 // DeleteApp deletes an app from the ArtifactMap
 func (am *ArtifactMap) DeleteApp(appGUID string) {
-	am.DeleteItem(ResourceTypeApp, appGUID, true)
+	am.DeleteItem(ResourceTypeApp, appGUID, ArtifactEntryCompareTypeID)
```
Comment (SeBBBe) [inline session/artifactmap.go:146]: Just incorrect variable naming? Appid != itemid

Comment (DnlLrssn) [reply session/artifactmap.go:146]: AppID is saved in entry.ID
ItemID is saved in entry.ItemID
so correct variable naming as far as I can see

Comment (SeBBBe) [reply session/artifactmap.go:146]: ArtifactEntryCompareTypeID implies item Id?

Comment (SeBBBe) [reply session/artifactmap.go:146]: Alright. Well, .ID is used in most other places to signify itemId so I guess that's why I thought it seemed off.

**Review by SeBBBe — APPROVED** [review]

File: session/artifactmap.go (hunk 1 above)
Comment (DnlLrssn) [reply session/artifactmap.go:146]: well previous code was
```
Lookup(resourcetype, lookup string, id bool)
```
where `id` mean app GUID (in this case)

Comment (DnlLrssn) [reply session/artifactmap.go:146]: entry.ID as artifactmap entry is always resource ID not item ID

Comment (DnlLrssn) [reply session/artifactmap.go:146]:
```
	ArtifactEntry struct {
		Name         string      `json:"name"`
		ID           string      `json:"resourceId"`
		ItemID       string      `json:"ID,omitempty"`

```

Comment (SeBBBe) [reply session/artifactmap.go:146]: That snippet shows the confusion very well 😆

Comment (DnlLrssn) [reply session/artifactmap.go:146]: yea noticed the json say ID for some reason lol... well should only become json for logs so should be ok

Comment (SeBBBe) [reply session/artifactmap.go:146]: The way I see it, the json is the way most places refer to the itemid vs resourceid and the internal variable names are what is used in the gopherciser

Comment (DnlLrssn) [reply session/artifactmap.go:146]: It used to be the resourceid only, the item id was a later addition.

**Pull request merged**